sklearn/gaussian_process/_tpr.py

```diff
@@ -0,0 +1,441 @@
+        (`return_cov=True`). Note that at most one of the two can be requested.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features) or list of object
+            Query points where the GP is evaluated.
+
+        return_std : bool, default=False
+            If True, the standard-deviation of the predictive distribution at
+            the query points is returned along with the mean.
+
+        return_cov : bool, default=False
+            If True, the covariance of the joint predictive distribution at
+            the query points is returned along with the mean.
+
+        return_tShape : bool, default=False
+            If True, the shape parameter of the predictive t distribution at
+            the query points is returned along with the mean.
+
+        return_tShapeMatrix : bool, default=False
+            If True, the shape parameter of the joint predictive t distribution
+            the query points is returned along with the mean.
+
+        Returns
+        -------
+        y_mean : ndarray of shape (n_samples,) or (n_samples, n_targets)
+            Mean of predictive distribution at query points.
+
+        y_std : ndarray of shape (n_samples,) or (n_samples, n_targets), optional
+            Standard deviation of predictive distribution at query points.
+            Only returned when `return_std` is True.
+
+        y_cov : ndarray of shape (n_samples, n_samples) or \
+                (n_samples, n_samples, n_targets), optional
+            Covariance of joint predictive distribution at query points.
+            Only returned when `return_cov` is True.
+
+        y_tShape : ndarray of shape (n_samples, n_samples) or \
+                (n_samples, n_samples, n_targets), optional
+            Shape of joint predictive t distribution at query points.
+            Only returned when `return_cov` is True.
+        """
+        if [return_std, return_cov, return_tShape, return_tShapeMatrix].count(True) > 1:
+            raise RuntimeError(
+                "At most one of return_std, return_cov, return_tShape or "
+                + "return_tShapeMatrix can be requested."
+            )
+
+        ### Spread may be either std or cov ###
+        if not any([return_std, return_cov, return_tShape, return_tShapeMatrix]):
+            return super().predict(X, return_std, return_cov)
+        elif return_cov or return_tShapeMatrix:  # Return a matrix:
+            y_mean, y_spread = super().predict(X, return_cov=True)
+        else:  # Return spread metric at points
+            y_mean, y_spread = super().predict(X, return_std=True)
+
+        ### Adjust depending on desired posterior ###
+        if self.n > 0 and (return_tShape or return_tShapeMatrix):
+            scailing_factor = (self.m_dis + self.v0 - 2) / self.v
+        elif self.n > 0 and (return_std or return_cov):
+            scailing_factor = (self.m_dis + self.v0 - 2) / (self.v - 2)
```

Comment:
This may get confusing so would be a good idea to put the reference to the paper for the Mahalanobis distance based correction for the desired posterior